xcode/Subconscious/Shared/Components/OmniboxView.swift

```diff
@@ -37,30 +42,80 @@ struct OmniboxView: View {
     }
 }
 
-extension OmniboxView {
-    init(
-        address: MemoAddress?,
-        defaultAudience: Audience
-    ) {
-        let audience = address?.toAudience() ?? defaultAudience
-        let title = address?.slug.description ?? ""
-        self.init(
-            icon: Image(audience: audience),
-            title: title
-        )
+/// Helper view that knows how to format slashlinks with or without petname.
+struct OmniboxSlashlinkView: View {
+    var petname: String?
+    var slug: String
+    
+    var body: some View {
+        HStack(spacing: 0) {
+            if let petname = petname {
+                Text(verbatim: "@\(petname)").fontWeight(.medium)
+                Text(verbatim: "/\(slug)")
+            } else {
+                Text(verbatim: slug)
+            }
+        }
+        .lineLimit(1)
     }
 }
 
 struct OmniboxView_Previews: PreviewProvider {
+    struct TappableOmniboxTestView: View {
+        var addresses: [MemoAddress] = [
+            MemoAddress.public(Slashlink("@ksr/red-mars")!),
+            MemoAddress.public(Slashlink("@ksr/green-mars")!),
+            MemoAddress.public(Slashlink("@ksr/blue-mars")!),
+            MemoAddress.local(Slug("mars")!),
+            MemoAddress.local(Slug("a-very-long-note-title-that-goes-on-and-on")!),
+            MemoAddress.public(Slashlink("@ksr/a-very-long-note-title-that-goes-on-and-on")!)
+        ]
+        
+        @State private var address: MemoAddress? = nil
+        var defaultAudience = Audience.local
+
+        var body: some View {
+            OmniboxView(address: address, defaultAudience: defaultAudience)
+                .onTapGesture {
+                    withAnimation {
+                        self.address = addresses.randomElement()
+                    }
+                }
```

Comment:
Question: in general on iOS should there be an "active" state for UI controls? i.e. change in color or opacity while the pointer is down but before it triggers a "tap" event.

There's no indication that I am tapping on the omnibox when I do it but obviously **this** instance is just a preview.

Reply:
There is a basic default tap state when run on-device (button dims when pressing). To me this feels right for this control.